core/blockchain.go

```diff
@@ -666,6 +666,8 @@ func (self *BlockChain) InsertChain(chain types.Blocks) (int, error) {
 			PutTransactions(self.chainDb, block, block.Transactions())
 			// store the receipts
 			PutReceipts(self.chainDb, receipts)
+			// Write map map bloom filters
+			WriteMipmapBloom(self.chainDb, block.NumberU64(), receipts)
```

Comment:
I'm not sure this is correct here. This branch is only accessed if you insert a block that is the new head of a canonical chain. If the block you just imported is *currently* a side chain, then you'll never write the bloom filter. For the above two `PutTransactions` and `PutReceipts` this is not a problem because the `reorg` takes care of invoking these ops when blocks get reorged. I think the correct place to put this is a bit up, right after `PutBlockReceipts` (https://github.com/obscuren/go-ethereum/blob/mipmap-bloom/core/blockchain.go#L649). That would ensure that irrelevant of the "canonical status" of the new block, the bloom filters will be updated.